src/Handler.fs

```diff
@@ -139,10 +140,20 @@ module Handler =
                 return Error (Panic ex)
         }
 
+    /// Extract header from response.
     let extractHeader<'T, 'TResult, 'TError> (header: string) (next: HttpFunc<_ ,_, 'TError>) (context: HttpContext) = task {
         let success, values = context.Response.Headers.TryGetValues header
         let values = if success then values else Seq.empty
 
         return! next { Request = context.Request; Response = Ok values }
     }
 
+    /// Use the given token provider to get a bearer token to use. This enables e.g. token refresh.
+    let withTokenProvider<'TResult, 'TError> (provider: CancellationToken -> Task<string option>) (next: HttpFunc<HttpResponseMessage, 'TResult, 'TError>) (ctx: HttpContext) = task {
```

Comment:
I'd name it "withBearerTokenProvider", since you are also handling cancellation tokens and I don't know if there anything else.

Reply:
Keeping the shorter name, but trying to be more clear in the comments. A provider is btw defined as an abstract factory factory class, but in FP everything is just another function 😃